gateway/src/service.rs

```diff
@@ -271,20 +271,31 @@ impl GatewayService {
 
     pub async fn iter_user_projects_detailed(
         &self,
-        account_name: AccountName,
+        account_name: &AccountName,
+        offset: u32,
+        limit: u32,
     ) -> Result<impl Iterator<Item = (ProjectName, Project)>, Error> {
-        let iter =
-            query("SELECT project_name, project_state FROM projects WHERE account_name = ?1")
-                .bind(account_name)
-                .fetch_all(&self.db)
-                .await?
-                .into_iter()
-                .map(|row| {
-                    (
-                        row.get("project_name"),
-                        row.get::<SqlxJson<Project>, _>("project_state").0,
-                    )
-                });
+        let mut query = QueryBuilder::new(
+            "SELECT project_name, project_state FROM projects WHERE account_name = ",
+        );
+        query.push_bind(account_name);
+        query.push(" LIMIT ").push_bind(limit);
+
+        if offset > 0 {
+            query.push(" OFFSET ").push_bind(offset);
```

Comment:
I just realized that we don't have an ORDER BY clause on this one, but this might be because there isn't something timestamp-wise we can order by. Do you know if we have guarantees that the query will return the rows in the same order though (and which one that might be)? I am thinking this can be an issue only if we query for a specific page based on a projects order and then query for the next page but we'll be basing it on another projects order.

Reply:
Indeed we shouldn't [rely on that](https://www.postgresql.org/docs/current/queries-order.html). Most of the time this should not be an issue, but let's not bet on that :)

> If sorting is not chosen, the rows will be returned in an unspecified order. The actual order in that case will depend on the scan and join plan types and the order on disk, but it must not be relied on. 

The best way would be to add a `created_at` column to the project table. Should we go with that solution ?

Reply:
Hey no worries ! 

The idea was to actually backfill the `created_at` field with the current timestamp. This won't be correct but since we got to start tracking it at some point. The issue with that solution is that we still need to solve ties between old projects. For that I actually found a way to generate a datetime for `created_at` that follows the disk order, while still having a different `created_at` value between each project. This should be the closest we can get from the actual insertion order, and we shouldn't get any tie after that. I'll go ahead and implement this, but please let me know if you see any problem with this solution.

Reply:
We've agreed to go with adding the `created_at` column for the next projects and sort only by the `project_name` for the old projects where we can not derive the `created_at` column in a sane manner. I'll keep this comment open to track this effort as well.